test/info_test.go

```diff
@@ -0,0 +1,61 @@
+package test
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/openfaas/faas-provider/types"
+	"github.com/openfaas/faas-swarm/handlers"
+)
+
+const (
+	infoTestVersion = "swarmtest"
+	infoTestSHA     = "test"
+)
+
+func TestMakeInfoHandler(t *testing.T) {
+	rr := httptest.NewRecorder()
+
+	req, err := http.NewRequest("GET", "/system/info", bytes.NewReader([]byte("")))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	handler := handlers.MakeInfoHandler(infoTestVersion, infoTestSHA)
+	infoRequest := types.InfoRequest{}
+
+	handler(rr, req)
+	body, err := ioutil.ReadAll(rr.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = json.Unmarshal(body, &infoRequest)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if required := http.StatusOK; rr.Code != required {
+		t.Errorf("handler returned wrong status code: got %v, but wanted: %v", rr.Code, required)
+	}
+
+	if infoRequest.Orchestration != handlers.SwarmIdentifier {
+		t.Errorf("handler returned wrong orchestration: got %v, but wanted: %v", infoRequest.Orchestration, handlers.SwarmIdentifier)
+	}
+
+	if infoRequest.Provider != handlers.SwarmProvider {
+		t.Errorf("handler returned wrong provider: got %v, but wanter: %v", infoRequest.Provider, handlers.SwarmProvider)
```

Comment:
There is a typo on this line `wanted`

Can we use: `etc.... - want: %v, got: %v` instead please?